klippy/gcode.py

```diff
@@ -63,6 +63,7 @@ def set_printer_ready(self, is_ready):
         self.heaters = [ e.get_heater() for e in extruders ]
         self.heaters.append(self.printer.objects.get('heater_bed'))
         self.fan = self.printer.objects.get('fan')
+        self.printer.objects.get('extruder_auto_fan').printer_ready=True
```

Comment:
I don't think gcode.py should be changed for this feature.  The klippy.py code that calls gcode.set_printer_ready() should be changed to also call a new fan.set_printer_ready() method.